src/components/input/Select.tsx

```diff
@@ -112,22 +126,37 @@ function SelectOption<T>({
         },
         classes,
       )}
-      onClick={() => {
-        if (!disabled) {
-          selectOrToggle();
+      onClick={e => {
+        // Do not invoke callback if clicked element is the checkbox, as it has
+        // its own event handler.
+        if (!disabled && e.target !== checkboxRef.current) {
+          selectOneValue();
         }
```

Comment:
I first tried to simply do `e.stopPropagation()` in the checkbox event handler, but it is an `onChange` handler that triggers after this `onClick`, so it didn't work.

That's why I did this explicit check here instead.